src/providers/extraction_providers/visionExtractionProvider.ts

```diff
@@ -0,0 +1,74 @@
+import { ExtractionProvider } from './extractionProvider';
+import { Image, ExtractedImage } from '../../models/data_models/image';
+import { AssignedLabel, DetectedObject, Coordinate } from '../../models/data_models/extraction';
+
+import * as vision from '@google-cloud/vision';
+
+const enum requestTypes {
+	labelDetection = 'LABEL_DETECTION',
+	objectLocalization = 'OBJECT_LOCALIZATION',
+}
+
+export class VisionExtractionProvider implements ExtractionProvider {
+	protected createAssignedLabel(
+		label: vision.protos.google.cloud.vision.v1.IEntityAnnotation
+	): AssignedLabel {
+		return {
+			description: label?.description || '',
+			confidence: label?.score || 0,
+		};
+	}
+
+	protected createBoundingPoly(
```

Comment:
do points count as polygons?
```suggestion
	protected createBoundingVertex(
```

Reply:
This utility function makes no use of `this`, so it could actually be made static

Reply:
does it make use of `this` on line 35?

Reply:
no, it's simply using `this` to access the function itself